### PR TITLE
RD-3242 Added edit mode support for page menu items icons

### PR DESCRIPTION
**Pull request by kubama**
## Description
Added edit mode support for page menu items icons

## Screenshots / Videos
image

## Checklist

- [X] My code follows the UI Code Style.
- [x] I verified that all tests and checks have passed.
- [X] I verified if that change requires a documentation update.
- [X] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/job/Stage-UI-System-Test/1442/

 

## Documentation
Will be updated in separate ticket

### CODE REVIEW

File: app/components/sidebar/IconSelection.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    const [currentValue, setCurrentValue, resetCurrentValue] = useInput(value);
+    const [opened, open, close] = useOpen(resetCurrentValue);
+
+    function handleSubmit() {
+        close();
+        onChange(currentValue);
+    }
+
+    return (
+        <Popup
```
Comment (qooban) [inline app/components/sidebar/IconSelection.tsx:23]: Did you consider extending `PopupConfirm` component and using it here instead of generic `Popup`?

**Review by qooban — COMMENTED** [review]
Added some comments.

Tested UI quickly, works fine 👍🏻 I think from the UX perspective it would be nice to indicate somehow that icon is editable (change style - change color or add border, maybe only on hover), but that can be done as well when we'll get to Edit Mode epic (ref. https://zpl.io/VqMlrkm).

File: app/components/sidebar/PagesList.tsx [hunk 2]
```diff
@@ -217,12 +232,21 @@ const PagesList: FunctionComponent<PagesListProps> = ({ isEditMode, pageId }) =>
...
+                              style={iconStyle}
+                              value={pageMenuItem.icon}
+                              onChange={icon => onIconChange(pageMenuItem.id, icon)}
+                          />
+                      )
+                    : pageMenuItem.icon && (
+                          <Icon
+                              name={pageMenuItem.icon}
+                              style={iconStyle}
+                              onClick={isEditMode ? () => {} : undefined}
```
[the remark this answers is not in the document]
Comment (kubama) [reply app/components/sidebar/PagesList.tsx]: Right, removed

File: app/components/sidebar/IconSelection.tsx (hunk 1 above)
Comment (kubama) [reply app/components/sidebar/IconSelection.tsx:23]: I did, unfortunately for some reason I was having issues with `SemanticIconDropdown` nested in it so I decided not to spend time on investigation and drop the idea.

Comment (qooban) [reply app/components/sidebar/IconSelection.tsx:23]: Got it. Makes sense to not spend time on it in that case.

**Review by qooban — DISMISSED** [review]
Some lint issues still needs to be fixed, but I have no more comments, so let me know when it's fixed, I'll re-approve.

**Review by qooban — APPROVED** [review]

**Pull request merged**